src/ADAL.PCL/Telemetry.cs

```diff
@@ -0,0 +1,126 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    public class Telemetry
+    {
+        private static readonly Telemetry Instance = new Telemetry();
+
+        private readonly IDictionary<Tuple<string, string>, DateTime> EventTracking =
+            new ConcurrentDictionary<Tuple<string, string>, DateTime>();
+
+        private readonly string format = "yyyy-mm-dd hh:mm:ss:ffff";
+
+        private DefaultAggregator Dispatcher;
+
+        public static Telemetry GetInstance()
+        {
+            return Instance;
+        }
+
+        internal string CreateRequestId()
+        {
+            return Guid.NewGuid().ToString();
+        }
+
+        public void RegisterDispatcher(IDispatcher dispatcher, bool aggregationRequired)
+        {
+            if (dispatcher == null)
+            {
+                throw new AdalException(AdalError.DispatcherIsNull);
```

Comment:
don't do this, setting dispatcher to null is currently the way to remove telemetry.